server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -60,6 +60,12 @@ private QueryPhase() {}
 
     public static void execute(SearchContext searchContext) throws QueryPhaseExecutionException {
         if (searchContext.queryPhaseRankShardContext() == null) {
+            if (searchContext.request().source() != null && searchContext.request().source().rankBuilder() != null) {
+                // if we have a custom Ranker (i.e. RankBuilder) provided, we want to fetch all
```

Comment:
nit: why custom? Any rank builder will apply this modif no?

Reply:
++ the `custom` is a bit confusing. Will remove it :)

Reply:
the only difference is with RRF (for now) as it also defines a `queryPhaseRankShardContext`